### PR TITLE
Homework 2 submission

### PR DESCRIPTION
**Pull request by charleslimjh**
API Documentation can be found in the README.md file.

### CODE REVIEW

File: Auth/AuthAPI.py [hunk 1]
```diff
@@ -1,7 +1,44 @@
...
+
+
+@auth_api.route("/register", methods=["POST"])
+def register():
+    if request.method == "POST":
+        req =  request.form
+
+        # checks for valid body
+        if "username" in req.keys() and "passwordHash" in req.keys():
+            Auth_db.append({"username": req["username"], "passwordHash": req["passwordHash"]})
```
Comment (nobelang78) [inline Auth/AuthAPI.py:18]: abit of defensive programming, what happens when passwordHash is empty string, or not present?

File: Auth/AuthAPI.py [hunk 2]
```diff
@@ -1,7 +1,44 @@
...
+    if request.method == "POST":
+        req = request.form
+
+        # checks for valid body
+        if "username" in req.keys() and "passwordHash" in req.keys():
+            username = req["username"]
+            passwordHash = req["passwordHash"]
+
+            token = jwt.encode({'username': username,'passwordHash': passwordHash}, current_app.config['SECRET_KEY'], algorithm='HS256')
+            return json.jsonify({"status": "success", "token": token})
```
Comment (nobelang78) [inline Auth/AuthAPI.py:34]: Perhaps my instruction were unclear but provided username + pw should be registered

File: README.md [hunk 3]
```diff
@@ -1,29 +1,266 @@
...
+```
+
+---
+
+
+# Auth  
+
+## POST /register<br>  
+
+Stores a username and hashedPassword in a local array.
```
Comment (nobelang78) [inline README.md:197]: Documentation should not provide information about implementation (local array) should simply tell the client what this does (registers a new account)

**Review by codecodecoder78 — COMMENTED** [review]
Overall great job. Couple of point to take note

1. Should use try-catch blocks to check for errors and return a specific error code eg. invalid input, unauthorised
2. Every return is currently returning 200, even on error

**Pull request closed without merging**